### PR TITLE
feat: add feedPlus directive

### PR DESCRIPTION
**Pull request by rebelchris**
New directive `@feedPlus`

It works basically same as `@auth` but when `feedId` argument is present it can validate if you are requesting my-feed or custom feed. If custom feed it will validate you are plus member.

Todo in separate ticket:
Add content Preference tests (need the feedId available there)

AS-835 #done

### CODE REVIEW

File: src/common/user.ts [hunk 1]
```diff
@@ -16,8 +16,8 @@ import {
   User,
   View,
 } from '../entity';
-import { ghostUser } from '../common';
-import { cancelSubscription } from '../common/paddle';
+import { ghostUser } from './index';
```
Comment (rebelchris) [inline src/common/user.ts:19]: This wasn't really a directive, not sure why it was there

File: __tests__/helpers.ts [hunk 2]
```diff
@@ -140,6 +140,7 @@ export const testMutationError = async (
   const res = await client.mutate(mutation.mutation, {
     variables: mutation.variables,
   });
+  console.log(res);
```
Comment (rebelchris) [inline __tests__/helpers.ts:143]:
```suggestion
```

**Review by idoshamun — APPROVED** [review]

File: src/schema/contentPreference.ts [hunk 3]
```diff
@@ -161,7 +161,7 @@ export const typeDefs = /* GraphQL */ `
       Follow status
       """
       status: FollowStatus!
-    ): EmptyResponse @auth
+    ): EmptyResponse @feedPlus
```
Comment (capJavert) [inline src/schema/contentPreference.ts:164]: I would still expect to use `@auth` as well?

File: src/directive/feedPlus.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        fieldConfig.resolve = function (source, args, ctx: Context, info) {
+          if (
+            !ctx.userId ||
+            (args?.feedId && ctx.userId !== args?.feedId && !ctx.isPlus)
+          ) {
+            if (['Query', 'Mutation'].includes(typeName)) {
+              throw new AuthenticationError(
+                'Access denied! You need to be authorized to perform this action!',
+              );
+            }
```
Comment (capJavert) [inline src/directive/feedPlus.ts:36]: separate in two IFs, one for `@auth` and second one which checks plus?

**Review by capJavert — APPROVED** [review]

**Pull request merged**